src/utils/git-ignore.ts

```diff
@@ -20,25 +23,23 @@ export function modifyGitIgnore(projectDir: string, addCds: boolean) {
     pathsToIgnore.push(...cdsPathsToIgnore);
   }
 
-  if (fs.existsSync(pathToGitignore)) {
-    try {
-      const fileContent = fs.readFileSync(pathToGitignore, 'utf8');
-      const newPaths = pathsToIgnore.filter(
-        filePath => !fileContent.includes(filePath)
-      );
-      const newFileContent =
-        fileContent + (newPaths.length ? `\n${newPaths.join('\n')}\n` : '');
+  try {
+    const fileContent = await readFile(pathToGitignore, 'utf8');
```

Comment:
Async functions are non-blocking which improves performance as parallel operations dont have to stop and wait for the readFile to finish. 

Actually I would have perferred to use [`fs/promises`](https://nodejs.org/api/fs.html#fs_promise_example) where all the functions are already wrapped with promisify out of the box, but it is only available in node14+.

Reply:
Sorry, my bad for the unclear question.
I actually would like to ask the difference between the old implementation `readFileSync` (blocking function) and the new one `await readFile()` (becomes blocking with await) which then calls `promisify(fs.readFile)`, since they are both blocking.

Reply:
Unfortunately, we can only get rid of node 12 in 2022.

Reply:
It is pretty similar but not 100% identical if I am not mistaken. But it might be a valid point to await only once the string is actually needed to allow the program to run a bit more in the background.

> Async/await makes your code look synchronous, and in a way it makes it behave more synchronously. The await keyword blocks execution of all the code that follows until the promise fulfills, exactly as it would with a synchronous operation. **It does allow other tasks to continue to run in the meantime, but your own code is blocked.** - [MDN](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await#:~:text=Async%2Fawait%20makes%20your%20code,would%20with%20a%20synchronous%20operation.)

Reply:
Ok, for this specific usage, we don't really see huge difference between these two.
I personally like this file `src/utils/fs.ts`, where we have all the file system related functions with the async manner, so you can easily choose whether you need await or not.